File: market/test/java/ru/yandex/market/wms/picking/modules/TestUserProvider.java
package ru.yandex.market.wms.picking.modules;

import java.util.HashSet;
import java.util.Set;

import org.apache.commons.lang.NotImplementedException;
import org.springframework.context.annotation.Primary;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;

import ru.yandex.market.wms.shared.libs.authorization.SecurityDataProvider;
import ru.yandex.market.wms.shared.libs.env.conifg.Profiles;

@Component
@Primary
@Profile({Profiles.TEST})
public class TestUserProvider implements SecurityDataProvider {

    @Override
    public String getUser() {
        return "anonymousUser";
    }

    @Override
    public String getToken() {
        throw new NotImplementedException();
    }

    @Override
    public Set<String> getRoles() {
        return new HashSet<>();
    }
}

File: market/test/kotlin/ru/yandex/market/wms/yt/provider/TestUserProvider.kt
package ru.yandex.market.wms.yt.provider

import org.apache.commons.lang.NotImplementedException
import org.springframework.context.annotation.Primary
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component
import ru.yandex.market.wms.shared.libs.env.conifg.Profiles

@Component
@Primary
@Profile(ru.yandex.market.wms.shared.libs.env.conifg.Profiles.TEST)
class TestUserProvider : ru.yandex.market.wms.shared.libs.authorization.SecurityDataProvider {
    override fun getUser(): String {
        return "anonymousUser"
    }

    override fun getToken(): String {
        throw NotImplementedException()
    }

    override fun getRoles(): Set<String> {
        return HashSet()
    }
}
